src/signature.c

```diff
@@ -300,6 +300,57 @@ static X509 *load_cert_file(const gchar *certfile, GError **error)
 	return res;
 }
 
+static STACK_OF(X509) *load_certs_from_file(const gchar *certfile, GError **error)
+{
+	BIO *cert_bio = NULL;
+	X509 *cert_x509 = NULL;
+	STACK_OF(X509) *certs = NULL;
+	unsigned long err;
+	const gchar *data;
+	int flags;
+
+	g_return_val_if_fail(certfile != NULL, NULL);
+	g_return_val_if_fail(error == NULL || *error == NULL, NULL);
+
+	cert_bio = BIO_new_file(certfile, "r");
+	if (cert_bio == NULL) {
+		g_set_error(
+				error,
+				R_SIGNATURE_ERROR,
+				R_SIGNATURE_ERROR_LOAD_FAILED,
+				"Failed to load cert file '%s'", certfile);
+		goto out;
+	}
+
+	cert_x509 = PEM_read_bio_X509(cert_bio, NULL, NULL, NULL);
+	if (cert_x509 == NULL) {
+		err = ERR_get_error_line_data(NULL, NULL, &data, &flags);
+		g_set_error(
+				error,
+				R_SIGNATURE_ERROR,
+				R_SIGNATURE_ERROR_PARSE_ERROR,
+				"Failed to parse cert file '%s': %s", certfile,
+				(flags & ERR_TXT_STRING) ? data : ERR_error_string(err, NULL));
+		goto out;
+	}
+
+	certs = sk_X509_new_null();
+	sk_X509_push(certs, cert_x509);
+
+	/* attempt to read further certfiles from PEM file */
+	while ((cert_x509 = PEM_read_bio_X509(cert_bio, NULL, NULL, NULL))) {
+		sk_X509_push(certs, cert_x509);
+	}
```

Comment:
If fear that this does not change anything as even in case of an error we might have reached the end of the file. e.g. when the last line of the file is broken.

I've reworked the method for proper error handling